src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimSpecCallbackHandler.java

```diff
@@ -325,4 +327,24 @@ private CallbackResponse validateRespondentPaymentDate(CallbackParams callbackPa
             .errors(errors)
             .build();
     }
+
+    private CallbackResponse validateLengthOfUnemployment(CallbackParams callbackParams) {
+
+        CaseData caseData = callbackParams.getCaseData();
+        List<String> errors = new ArrayList<>();
+
+        if (caseData.getRespondToClaimAdmitPartUnemployedLRspec() != null
+            && caseData.getRespondToClaimAdmitPartUnemployedLRspec().getLengthOfUnemployment() != null) {
+            if (caseData.getRespondToClaimAdmitPartUnemployedLRspec().getLengthOfUnemployment()
+                .getNumberOfYearsInUnemployment().contains(".")
+                || caseData.getRespondToClaimAdmitPartUnemployedLRspec()
+                .getLengthOfUnemployment().getNumberOfMonthsInUnemployment().contains(".")) {
+                errors.add("Length of time unemployed must be whole numbers, like 10.");
```

Comment:
I think that the if would be more clear with a method unemployedHasFraction(caseData) {
  if (caseData.respondToClaimAdmitPartUnemployedLRspec != null && 
  caseData.respondToClaimAdmitPartUnemployedLRspec.lengthOfUnemployment != null) {
    return caseData.respondToClaimAdmitPartUnemployedLRspec.lengthOfUnemployment.numberOfYearsInUnemployment.contains(".")
       || caseData.respondToClaimAdmitPartUnemployedLRspec.lengthOfUnemployment.numberOfMonthsInUnemployment.contains(".")
  }
}

Then the if becomes if (unemployedHasFraction(caseData)) {...}

The method's body could also be 
return Optional.ofNullable(caseData.getRespondToClaimAdmitPartUnemployedLRspec()).map(CLASS::getLengthOfUnemployment).map(length -> 
  length.numberOfYearsInUnemployment.contains(".") || length.numberOfMonthsInUnemployment.contains(".")
)

if that is easier to read

Reply:
This was contained in here as part of another ticket CIV-232 changes (which is now merged). Now it does not show up here. I have noted it as part of checking it separately

Reply:
I have informed owner of CIV-232, if we can consider these changes in a separate refactor ticket.

Reply:
don't understand how this is showing in the difference from the first place? it should have been already part of master. but yes like @bhssamray suggested if this change is needed we need a separate ticket for it